core/src/test/java/org/elasticsearch/percolator/PercolatorQueryTests.java

```diff
@@ -0,0 +1,168 @@
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+public class PercolatorQueryTests extends ESTestCase {
+
+    private Directory directory;
+    private IndexWriter indexWriter;
+    private DirectoryReader directoryReader;
+
+    @Before
+    public void init() throws Exception {
+        directory = newDirectory();
+        IndexWriterConfig config = new IndexWriterConfig(new WhitespaceAnalyzer());
+        config.setMergePolicy(NoMergePolicy.INSTANCE);
+        indexWriter = new IndexWriter(directory, config);
+    }
+
+    @After
+    public void destroy() throws Exception {
+        directoryReader.close();
+        directory.close();
+    }
+
+    public void testVariousQueries() throws Exception {
+        Map<BytesRef, Query> queries = new HashMap<>();
+        QueryMetadataService queryMetadataService = new QueryMetadataService();
+        addPercolatorQuery("1", new TermQuery(new Term("field", "brown")), indexWriter, queryMetadataService, queries);
+        addPercolatorQuery("2", new TermQuery(new Term("field", "monkey")), indexWriter, queryMetadataService, queries);
+        addPercolatorQuery("3", new TermQuery(new Term("field", "fox")), indexWriter, queryMetadataService, queries);
+        BooleanQuery.Builder bq1 = new BooleanQuery.Builder();
+        bq1.add(new TermQuery(new Term("field", "fox")), BooleanClause.Occur.SHOULD);
+        bq1.add(new TermQuery(new Term("field", "monkey")), BooleanClause.Occur.SHOULD);
+        addPercolatorQuery("4", bq1.build(), indexWriter, queryMetadataService, queries);
+        BooleanQuery.Builder bq2 = new BooleanQuery.Builder();
+        bq2.add(new TermQuery(new Term("field", "fox")), BooleanClause.Occur.MUST);
+        bq2.add(new TermQuery(new Term("field", "monkey")), BooleanClause.Occur.MUST);
+        addPercolatorQuery("5", bq2.build(), indexWriter, queryMetadataService, queries);
+        BooleanQuery.Builder bq3 = new BooleanQuery.Builder();
+        bq3.add(new TermQuery(new Term("field", "fox")), BooleanClause.Occur.MUST);
+        bq3.add(new TermQuery(new Term("field", "apes")), BooleanClause.Occur.MUST_NOT);
+        addPercolatorQuery("6", bq3.build(), indexWriter, queryMetadataService, queries);
+        BooleanQuery.Builder bq4 = new BooleanQuery.Builder();
+        bq4.add(new TermQuery(new Term("field", "fox")), BooleanClause.Occur.MUST_NOT);
+        bq4.add(new TermQuery(new Term("field", "apes")), BooleanClause.Occur.MUST);
+        addPercolatorQuery("7", bq4.build(), indexWriter, queryMetadataService, queries);
+        PhraseQuery.Builder pq1 = new PhraseQuery.Builder();
+        pq1.add(new Term("field", "lazy"));
+        pq1.add(new Term("field", "dog"));
+        addPercolatorQuery("8", pq1.build(), indexWriter, queryMetadataService, queries);
+
+        indexWriter.close();
+        directoryReader = DirectoryReader.open(directory);
+        IndexSearcher shardSearcher =  new IndexSearcher(directoryReader);
```

Comment:
Can you do "newSearcher" instead of "new IndexSearcher" in these tests so that we get an AssertingIndexSearcher which might catch some bugs?